src/youtube-player/youtube-player.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  ViewChild,
+} from '@angular/core';
+
+import {
+  combineLatest,
+  of,
+  Observable,
+  Subscription,
+  ConnectableObservable,
+  pipe,
+  MonoTypeOperatorFunction,
+  merge,
+  OperatorFunction,
+} from 'rxjs';
+
+import {
+  map,
+  scan,
+  withLatestFrom,
+  flatMap,
+  filter,
+  startWith,
+  publish,
+  first,
+  distinctUntilChanged,
+} from 'rxjs/operators';
+
+declare global {
+  interface Window { YT: typeof YT; }
+}
+
+// The player isn't fully initialized when it's constructed.
+// The only field available is destroy.
+type UninitializedPlayer = Pick<YT.Player, 'destroy'| 'addEventListener'>;
+
+/**
+ * Simple component that previews a YouTube video.
+ */
+@Component({
+  selector: 'youtube-player',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: 'youtube-player.html',
+})
+export class YouTubePlayer implements AfterViewInit, OnDestroy {
+  /** YouTube Video ID to view */
+  @Input() set videoId(videoId: string|undefined) {
+    this._videoId$.emit(videoId);
+  }
+  private _videoId$ = new EventEmitter<string|undefined>();
+
+  /** The element that will be replaced by the iframe. */
+  @ViewChild('youtube_container', {static: false}) youtubeContainer: ElementRef|undefined;
+  private _youtubeContainer$ = new EventEmitter<HTMLElement>();
+
+  private _player: YT.Player|undefined;
+  private _subscriptions: Subscription[] = [];
```

Comment:
Rather than tracking subscription, we prefer to introduce a `_destroyed` subject, use `takeUntil(this._destroyed)`, and emit `_destroyed` in `ngOnDestroy`